### DISCUSSION TITLE
WebView overlay scrollbars

### DISCUSSION THREAD

**Discussion by matthewjamesadam** [microsoft/vscode-discussions#998] [Extension Development QnA] [answered]
Hello friends,

VSCode's built-in scrollbars are `overlay` scrollbars: scrollbars overlay the content displayed in the view, and are only shown while the mouse is over the view.  You can see the effect here, in the file explorer:

https://github.com/microsoft/vscode-discussions/assets/2133518/ab78f9d9-092b-41a8-bfa9-34a3aa8592ac

However, any scrollbars in a WebView are fixed scrollbars.  They are always visible, and crucially they always take up space on the right-hand side of the WebView.  See this with the calico colors sample app:

https://github.com/microsoft/vscode-discussions/assets/2133518/42fa3ab8-6ae2-4e90-a8b3-66fccb5a8d90

If an extension's WebViews have any full-width content, the end effect with scrollbars is visually unpleasant.  There will always be gaps and scroll UI on the right-hand side.

Is there a reason the scrollbars in WebViews are different then the scrollbars built into native parts of VSCode?  And can this be changed?  I managed to apply some `-webkit-scrollbar-*` styling in my webview that mitigated some of this, but I cannot find a way to make the scrollbars overlay content correctly.

If this is something requiring a VSCode change, I would be happy to look into making the change myself, if I could get some guidance as to where the relevant code is.

Thanks!

**Comment by tamuratak** [comment, marked as answer]
> Is there a reason the scrollbars in WebViews are different then the scrollbars built into native parts of VSCode?

The scrollbars built into native parts of VSCode is implemented with the `div` element, `<div class="slider"></div>`. You can see it with the `Developer: Toggle Developer Tools` command.

> I managed to apply some -webkit-scrollbar-* styling in my webview that mitigated some of this, but I cannot find a way to make the scrollbars overlay content correctly.

You can refer to #506.

Another implementation is GitLens' Commit Details WebView: https://github.com/gitkraken/vscode-gitlens/blob/b1d71d4844523e8b2ef16f9e007068e91f46fd88/src/webviews/apps/home/home.scss#L55-L104
